docs/ci/ci-automated-release-workflow.md

```diff
@@ -0,0 +1,134 @@
+# Creating Automated Releases: Design Document
+
+## Motivation & Overview
+
+Presently, the release processes for every library within this `instructlab` GitHub organization is entirely manual:
+
+For example, for a typical y-stream release, a maintainer has to:
+
+1. Manually create a new release branch -- e.g., `release-0.y.0`,
+2. Manually create a pull request against `release-0.y.0` to cap the versions on some of the dependencies defined in their library's `requirements.txt` file,
```

Comment:
Where is this requirement to cap coming from? Aren't we asked to [not](https://github.com/instructlab/dev-docs/blob/main/docs/dependency-management.md) cap?

```
Only apply "caps" to dependencies (using <) when that dependency has established a pattern of producing new releases with breaking changes.
```

Without this, steps 3-5 are no longer necessary.

Reply:
We have been capping certain dependencies in the core repo since October 2024: https://github.com/instructlab/instructlab/pulls?q=is%3Apr+%22deps%3A+cap%22+is%3Aclosed+

To elaborate, we have been capping our own InstructLab library dependencies to ensure that if we create a new release from the core repo, that new release won't automatically consume potential breaking changes from one of our own libraries.

Reply:
Also, to be clear, the "automatic capping" feature is not required to be used by anybody. Maintainers can ignore the feature if desired. And since the automation described here will create a pull request that someone has to manually approve and review, the dependency cappping changes not be merged automatically. 😃